spacy/pipeline/edit_tree_lemmatizer.py

```diff
@@ -160,27 +160,18 @@ def predict(self, docs: Iterable[Doc]) -> List[Ints2d]:
     def _scores2guesses(self, docs, scores):
         guesses = []
         for doc, doc_scores in zip(docs, scores):
-            if self.top_k == 1:
-                doc_guesses = doc_scores.argmax(axis=1).reshape(-1, 1)
-            else:
-                doc_guesses = np.argsort(doc_scores)[..., : -self.top_k - 1 : -1]
-
-            if not isinstance(doc_guesses, np.ndarray):
-                doc_guesses = doc_guesses.get()
-
             doc_compat_guesses = []
-            for token, candidates in zip(doc, doc_guesses):
-                tree_id = -1
-                for candidate in candidates:
+            for i, token in enumerate(doc):
+                for _ in range(self.top_k):
```

Comment:
One obvious question here is what happens if the value of `top_k` is greater than the number of trees in the model. It doesn't seem worth testing for this explicitly:

- with genuine models and genuinely useful values of `top_k`, it's not going to happen (a typical models has more than a thousand trees, a high useful value of `top_k` might be 10 or 20).
- Any real model will contain a tree that does not apply any change to the original form, which by definition can be applied to any token, so that trees will never be looped through a second time.
- In the exceedingly unlikely event that for some reason all the trees in a model were not applicable to a given token form, nothing would break: the first tree in the NumPy array would just be tried repeatedly until the value of `top_k` were reached.
- Testing for this explicitly would however have a real-world speed impact.

Reply:
I don't follow. The number of labels is `len(self.labels)`, so this would be a single `range(min(self.top_k, len(self.labels))`. Maybe I am missing something?

Reply:
OK, no, you're right — not really a real-world speed impact, but still extra processing: do you think this is worth doing given how unlikely it is to ever happen and the fact that nothing would break if it did?

Reply:
Yes, it’s the semantics that it should convey and if it is cheap to be correct, why not?